:sparkles: Add history feature

matissecallewaert opened the pull request:
Add all previously picked images in a scrollable. 
Add a toggle to enable or disable the history feature. 
Cleanup and format of the code.

[Screencast from 2024-12-05 22:37:13.webm](https://github.com/user-attachments/assets/7d05db5c-501a-4991-953f-7c9cfdc8d2d0)

matissecallewaert committed:
:sparkles: Add history feature

ManuDeBuck commented on index.html:
```diff
@@ -156,6 +156,20 @@ <h1 class="mt-3 mb-3">Random Image Picker</h1>
                         <img id="random-image" class="img-thumbnail random-image" src="" alt="Randomly picked image.">
                     </div>
                 </div>
+                <div class="col-lg-4 mb-4 text-center">
+                    <div class="row ml-5">
+                        <h2 class="mt-3 mb-4">History:</h2>
+                        <div class="col-12 mb-4 history-image-wrapper">
+                            <img id="previous-image-1" class="img-thumbnail history-image" src="" alt="Previously picked image 1" style="display: none;">
+                        </div>
+                        <div class="col-12 mb-4 history-image-wrapper">
+                            <img id="previous-image-2" class="img-thumbnail history-image" src="" alt="Previously picked image 2" style="display: none;">
+                        </div>
+                        <div class="col-12 mb-4 history-image-wrapper">
+                            <img id="previous-image-3" class="img-thumbnail history-image" src="" alt="Previously picked image 3" style="display: none;">
+                        </div>
+                    </div>
+                </div>
```
I would propose to have all extra features in a feature toggle on the home screen. E.g. `show directly`.

ManuDeBuck replied:
Another option would be to allow for the default view (everyone is used to), and hide the history by default. Allowing people to show the history by, e.g. a window in the right side or so.

ManuDeBuck commented on index.html:
```diff
@@ -156,6 +156,20 @@ <h1 class="mt-3 mb-3">Random Image Picker</h1>
                 </div>
+                <div class="col-lg-4 mb-4 text-center">
+                    <div class="row ml-5">
+                        <h2 class="mt-3 mb-4">History:</h2>
```
```suggestion
                        <h2 class="mt-3 mb-4">Previous images</h2>
```

ManuDeBuck commented on index.html:
```diff
@@ -156,6 +156,20 @@ <h1 class="mt-3 mb-3">Random Image Picker</h1>
+                <div class="col-lg-4 mb-4 text-center">
+                    <div class="row ml-5">
+                        <h2 class="mt-3 mb-4">History:</h2>
+                        <div class="col-12 mb-4 history-image-wrapper">
+                            <img id="previous-image-1" class="img-thumbnail history-image" src="" alt="Previously picked image 1" style="display: none;">
+                        </div>
+                        <div class="col-12 mb-4 history-image-wrapper">
+                            <img id="previous-image-2" class="img-thumbnail history-image" src="" alt="Previously picked image 2" style="display: none;">
+                        </div>
+                        <div class="col-12 mb-4 history-image-wrapper">
+                            <img id="previous-image-3" class="img-thumbnail history-image" src="" alt="Previously picked image 3" style="display: none;">
+                        </div>
```
What do you think about a scrollable where you can see all previous images?

ManuDeBuck commented on actions.js:
```diff
@@ -100,11 +102,34 @@ function setFinalImage(index, deleteImage) {
 }
 
+function updatePickedImages() {
+    roundsPlayed += 1;
+    if (roundsPlayed >= 1) {
+        $("#previous-image-1").prop("src", pickedImageHistory[pickedImageHistory.length - 1]).css("display", "");
+    }
+    if (roundsPlayed >= 2) {
+        $("#previous-image-2").prop("src", pickedImageHistory[pickedImageHistory.length - 2]).css("display", "");
+    }
+    if (roundsPlayed >= 3) {
+        $("#previous-image-3").prop("src", pickedImageHistory[pickedImageHistory.length - 3]).css("display", "");
+    }
```
the variable roundsPlayed is equal to the pickedImageHistory.length. Seems like a better idea. The idea below (of having all previous images displayed), by the way, would allow to do this with a for-loop instead.

ManuDeBuck commented on actions.js:
```diff
@@ -1,4 +1,6 @@
 let IMAGES = []; // Will put all images in this
+let pickedImageHistory = []; // Will put all picked images in this
+let roundsPlayed = 0; // Amount of rounds played
```
I would propose to remove this variable, since it's equal to pickedImageHistory's length, right?

ManuDeBuck commented on index.html:
```diff
@@ -156,6 +156,20 @@ <h1 class="mt-3 mb-3">Random Image Picker</h1>
                 </div>
+                <div class="col-lg-4 mb-4 text-center">
+                    <div class="row ml-5">
+                        <h2 class="mt-3 mb-4">History:</h2>
```
I would propose to have this title styled a bit better. I know The application doesn't have the nicest UI, but still... 😉

matissecallewaert replied:
Any suggestions, I'm not that good with design XD

matissecallewaert committed:
:art: Update history feature

Add all previously picked images in a scrollable. Add a toggle to enable or disable the history feature. Cleanup and format of the code

matissecallewaert committed:
:bug: Fix display override by d-flex

ManuDeBuck committed:
:memo: Add note to mention history is empty

ManuDeBuck committed:
:lipstick: Add some visual tweaks, also for mobile friendlyness
